core/src/sumeragi/fault.rs

```diff
@@ -1305,500 +872,511 @@ pub fn run_sumeragi_main_loop<F>(
                         }
-                        Message::BlockSigned(block_signed) => {
-                            println!("block signed message");
-                            let block = block_signed.block;
-                            let block_hash = block.hash();
 
-                            if voting_block_option.is_some()
-                                && block_hash != voting_block_option.as_ref().unwrap().block.hash()
-                            {
-                                println!("block signed is not relevant block");
-                                continue;
-                            }
+                        let voting_block = VotingBlock::new(block.clone());
+                        voting_block_option = Some(voting_block);
 
-                            // I don't think we update the topology here. That ship has sailed.
+                        instant_at_which_we_should_have_committed =
+                            Instant::now() + sumeragi.commit_time;
+                    }
+                    Message::BlockSigned(block_signed) => {
+                        let block = block_signed.block;
+                        let block_hash = block.hash();
 
-                            let network_topology = &state_machine_guard.current_topology;
+                        if voting_block_option.is_some()
+                            && block_hash != voting_block_option.as_ref().unwrap().block.hash()
+                        {
+                            error!("block signed is not relevant block");
+                            continue;
+                        }
 
-                            let valid_signatures = network_topology.filter_signatures_by_roles(
+                        let valid_signatures = state_machine_guard
+                            .current_topology
+                            .filter_signatures_by_roles(
                                 &[Role::ValidatingPeer, Role::Leader],
                                 block.verified_signatures(),
                             );
 
-                            for sig in &valid_signatures {
-                                block_signature_acc.push((block_hash, sig.clone()));
-                            }
-                        }
-                        _ => {
-                            println!("Not handling message {:?}", incoming_message);
+                        for sig in &valid_signatures {
+                            block_signature_acc.push((block_hash, sig.clone()));
                         }
                     }
-                } else {
-                    // if there is no message sleep
-                    should_sleep = true;
+                    _ => {
+                        trace!("Not handling message {:?}", incoming_message);
+                    }
                 }
+            } else {
+                // if there is no message sleep
+                should_sleep = true;
+            }
```

Comment:
We have this branch in many places could we extract it somehow in single place?

Reply:
Not really because then you would have to link to that single place from all the places this bool is currently set. The problem is the problem, there is nothing we can do that makes the problem go away or become simpler than it actually is.